pallets/schemas/src/migration/v4.rs

```diff
@@ -1,137 +1,108 @@
-	pub struct OldSchemaInfo {
-		/// The type of model (AvroBinary, Parquet, etc.)
-		pub model_type: ModelType,
-		/// The payload location
-		pub payload_location: PayloadLocation,
-		/// additional control settings for the schema
-		pub settings: SchemaSettings,
+pub fn get_known_schemas<T: Config>() -> BTreeMap<SchemaId, Vec<u8>> {
+	let genesis_block: BlockNumberFor<T> = 0u32.into();
+	let genesis = <frame_system::Pallet<T>>::block_hash(genesis_block);
+
+	if let DetectedChainType::FrequencyMain = get_chain_type_by_genesis_hash(&genesis.encode()[..])
+	{
+		// only return what needs to change which for this case is only on mainnet
+		BTreeMap::from([
+			(5, b"dsnp.update".to_vec()),
+			(6, b"dsnp.profile".to_vec()),
+			(7, b"dsnp.public-key-key-agreement".to_vec()),
+			(8, b"dsnp.public-follows".to_vec()),
+			(9, b"dsnp.private-follows".to_vec()),
+			(10, b"dsnp.private-connections".to_vec()),
+		])
+	} else {
+		BTreeMap::new()
 	}
-
-	/// Old Storage for schema info struct data
-	#[storage_alias]
-	pub(crate) type SchemaInfos<T: Config> =
-		StorageMap<Pallet<T>, Twox64Concat, SchemaId, OldSchemaInfo, OptionQuery>;
 }
 
-/// migration to v3 implementation
-pub struct MigrateToV3<T>(PhantomData<T>);
+/// migration to v4 implementation
+pub struct MigrateToV4<T>(PhantomData<T>);
 
-impl<T: Config> OnRuntimeUpgrade for MigrateToV3<T> {
+impl<T: Config> OnRuntimeUpgrade for MigrateToV4<T> {
 	fn on_runtime_upgrade() -> Weight {
-		migrate_to_v3::<T>()
+		migrate_to_v4::<T>()
 	}
 
 	#[cfg(feature = "try-runtime")]
 	fn pre_upgrade() -> Result<Vec<u8>, TryRuntimeError> {
 		log::info!(target: LOG_TARGET, "Running pre_upgrade...");
 		let on_chain_version = Pallet::<T>::on_chain_storage_version();
-		if on_chain_version >= 3 {
+		let genesis_block: BlockNumberFor<T> = 0u32.into();
+		let genesis = <frame_system::Pallet<T>>::block_hash(genesis_block);
+		if on_chain_version >= 4 {
 			return Ok(Vec::new())
 		}
-		let count = old::SchemaInfos::<T>::iter().count() as u32;
-		log::info!(target: LOG_TARGET, "Finish pre_upgrade for {:?}", count);
-		Ok(count.encode())
+		log::info!(target: LOG_TARGET, "Found genesis... {:?}", genesis);
+		let detected_chain = get_chain_type_by_genesis_hash(&genesis.encode()[..]);
+		log::info!(target: LOG_TARGET,"Detected Chain is {:?}", detected_chain);
+		Ok(Vec::new())
 	}
 
 	#[cfg(feature = "try-runtime")]
 	fn post_upgrade(_: Vec<u8>) -> Result<(), TryRuntimeError> {
 		log::info!(target: LOG_TARGET, "Running post_upgrade...");
 		let on_chain_version = Pallet::<T>::on_chain_storage_version();
-		if on_chain_version >= 3 {
+		if on_chain_version >= 4 {
 			return Ok(())
 		}
 		let onchain_version = Pallet::<T>::on_chain_storage_version();
 		assert_eq!(onchain_version, SCHEMA_STORAGE_VERSION);
+		// check to ensure updates took place
+		let known_schemas = get_known_schemas::<T>();
+		for (schema_id, schema_name) in known_schemas.into_iter() {
+			// safe to use unwrap since only used in try_runtime
```

Comment:
Curious nit: why doesn't lint complain about unwrap here? I don't see it disabled at the top of the file?